pkg/api/v20230401/openshiftcluster_validatestatic.go

```diff
@@ -346,6 +365,24 @@ func (sv openShiftClusterStaticValidator) validateIngressProfile(path string, p
 	return nil
 }
 
+func (sv openShiftClusterStaticValidator) validateClusterResourceGroupTags(path string, t *Tags) error {
+	if len(*t) > maxTags {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, fmt.Sprintf("The provided set of cluster resource group tags is too large; it can contain at most %d tags.", maxTags))
+	}
+
+	for k, v := range *t {
+		if !sv.clusterResourceGroupTagIsValid(k, v) {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, path, "One or more of the provided cluster resource group tags are invalid.")
+		}
+	}
+
+	return nil
+}
+
+func (sv openShiftClusterStaticValidator) clusterResourceGroupTagIsValid(key string, value string) bool {
```

Comment:
if this returned the invalid tag name in addition or in replace of the bool we could then let the customer know which tag they provided was invalid.